internal/cmd/add_token.go

```diff
@@ -1,23 +1,176 @@
 package cmd
 
 import (
-	"fmt"
-	"os"
+    "bufio"
+    "fmt"
+    "io/ioutil"
+    "net/url"
+    "os"
+    "os/exec"
+    "path/filepath"
+    "runtime"
+    "strings"
+    "github.com/urfave/cli/v2"
+    s "github.com/yitsushi/totp-cli/internal/storage"
+    "github.com/yitsushi/totp-cli/internal/terminal"
+)
+
+func promptForInput(prompt, defaultValue string) string {
+    reader := bufio.NewReader(os.Stdin)
+    if defaultValue != "" {
+        prompt = fmt.Sprintf("%s (%s)", prompt, defaultValue)
+    }
+    fmt.Print(prompt + ": ")
+    text, err := reader.ReadString('\n')
+    if err != nil {
+        fmt.Fprintf(os.Stderr, "Error reading input: %v\n", err)
+        os.Exit(1)
+    }
+    text = strings.TrimSpace(text)
+    if text == "" {
+        text = defaultValue
+    }
+    return text
+}
 
-	"github.com/urfave/cli/v2"
 
-	s "github.com/yitsushi/totp-cli/internal/storage"
-	"github.com/yitsushi/totp-cli/internal/terminal"
-)
+func installZbarimg() error {
+    var cmd *exec.Cmd
+
+    switch os := runtime.GOOS; os {
+    case "linux":
+        cmd = exec.Command("sudo", "apt-get", "install", "zbar-tools")
```

Comment:
That's a huge no. Just as a starter, I use Arch on one machine, and Manjaro on another one.

My view on thins, a standalone software should NEVER install anything unless that's the purpose of the job.

Reply:
Thanks a LOT @yitsushi. Actually, I didn't really consider these factors, I was just concerned with it working as soon as I could lol 

I've learnt a lot about good software practices, thank you.